docs/fastdds/dds_layer/topic/typeSupport/typeSupport.rst

```diff
@@ -1,4 +1,71 @@
-.. _dds_layer_topic_typeSupport:
+.. _dds_layer_definition_data_types:
+
+Definition of data types
+========================
+
+The definition of the data type exchanged in a :ref:`dds_layer_topic_topic` is divided in
+two classes: the TypeSupport and the TopicDataType.
+
+TopicDataType describes the data type exchanged between a publication and a subscription, i.e.,
+the data corresponding to a :ref:`dds_layer_topic_topic`.
+The user has to create a specialized class for each concrete type that will be used by the application.
+
+Any specialization of TopicDataType must be registered in the :ref:`dds_layer_domainParticipant`
+before it can be used to create :ref:`dds_layer_topic_topic` objects.
+A TypeSupport object encapsulates an instance of TopicDataType, providing the methods needed to
+register the type and interact with the publication and subscription.
+To register the data type, create a new TypeSupport with a TopicDataType instance and use the :func:`register_type`
+method on the TypeSupport.
+Then the :ref:`dds_layer_topic_topic` can be created with the registered type name.
+
+.. note::
+
+   Registering two different data types on the same :ref:`dds_layer_domainParticipant` with identical names is not
+   allowed and will issue an error.
+   However, it is allowed to register the same data type within the same :ref:`dds_layer_domainParticipant` with
+   the same or different names.
+   If the same  the same data type is registered twice on the same :ref:`dds_layer_domainParticipant` with the same
+   name, the second registering will have no effect, but will not issue any error.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_TYPE_REGISTER
+   :end-before: //!
+
+
+.. _dds_layer_topic_dynamic_data_types:
+
+Dynamic data types
+------------------
+
+Instead of directly writing the specialized TopicDataType class, it is possible to dynamically define data types
+following the OMG Extensible and Dynamic Topic Types for DDS interface.
+Data types can also be described on an XML file that is dynamically loaded.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+   :language: c++
+   :start-after: //DDS_DYNAMIC_TYPES
+   :end-before: //!
+
+A complete description of the dynamic definition of types can be found on the section about :ref:`dynamic-types`.
+
+
+.. _dds_layer_topic_keyed_data_types:
+
+Keyed data types
```

Comment:
Key data types perhaps would be better because I haven't found the "keyed" word in the code or in the standard.

Reply:
Key data types sounds like 'data types that are key or important'. I'll change that to 'data types with key'